lib/service/service.go

```diff
@@ -3748,17 +3692,48 @@ func (process *TeleportProcess) initDiagnosticService() error {
 }
 
 // initDebugService starts debug service serving endpoints used for
-// troubleshooting the instance.
-func (process *TeleportProcess) initDebugService() error {
+// troubleshooting the instance. This service is always active, users can
+// disable its sensitive pprof and log-setting endpoints, but the liveness
+// and readiness ones are always active.
+func (process *TeleportProcess) initDebugService(exposeDebugRoutes bool) error {
+	if process.state == nil {
+		return trace.BadParameter("teleport process state machine has not yet been initialized (this is a bug)")
+	}
+
 	logger := process.logger.With(teleport.ComponentKey, teleport.Component(teleport.ComponentDebug, process.id))
 
+	// Unix socket creation can fail on paths too long. Depending on the UNIX implementation,
+	// socket paths cannot exceed 104 or 108 chars.
 	listener, err := process.importOrCreateListener(ListenerDebug, filepath.Join(process.Config.DataDir, teleport.DebugServiceSocketName))
+	if err != nil {
+		// If the debug service is enabled in the config, this is a hard failure
+		if exposeDebugRoutes {
+			return trace.Wrap(err)
+			// If the debug service was disabled in the config, we issue a warning and will have to continue.
+		} else {
+			logger.WarnContext(process.ExitContext(),
```

Comment:
style: consistent indentation

```suggestion
		if exposeDebugRoutes {
			// If the debug service is enabled in the config, this is a hard failure
			return trace.Wrap(err)
		} else {
			// If the debug service was disabled in the config, we issue a warning and will have to continue.
			logger.WarnContext(process.ExitContext(),
```